### DISCUSSION TITLE
Ability to modify notification message structure

### DISCUSSION THREAD

**Discussion by aliartiza75** [fluxcd/flux2#3043] [Proposals] [answered]
Hi,

I have recently enabled GitOps using flux. Everything is working fine. I push notifications to a google chat space, but the notification message structure is not expressive. 

I think users should be able to modify the message structure.

**Comment by makkes** [comment, marked as answer]
Thanks for your feedback. Have you looked at the `Alert` spec and particularly the `summary` field? Would you mind going into the specifics of what you would like to see changed?

**Reply by aliartiza75** [reply]
@makkes thank you for the response. Yes, I have looked at the summary field and I am using it to differentiate notifications/alerts. But I am thinking of making other fields configurable, i.e **Revision** and **artifacts** information, and maybe adding additional fields that are relevant to a specific type of notification.

**Reply by makkes** [reply]
This sounds a lot like a feature request for templating Alerts. 🤔 It's an interesting thought. How would you go about processing those additional fields?

**Reply by aliartiza75** [reply]
@makkes i would like to have a generic list of fields that a user can choose from and list down all the fields in the **Alert** definition. Like we specify eventSources. But it just an idea for now. I will go through the source code and will get back to you gain.

Secondly, It would be great to list down all the events that are captured via GitRepository, Kustomization eventSource.

**Comment by kingdonb** [comment]
There have been a few discussions about templating alerts, it has not made its way onto the roadmap yet. Thanks for the feedback!
